util/logutil/log.go

```diff
@@ -96,6 +96,25 @@ const (
 // SlowQueryLogger is used to log slow query, InitLogger will modify it according to config file.
 var SlowQueryLogger = log.L()
 
+// GeneralLogLogger is used to log general log
+var GeneralLogLogger = log.L()
+
+var generalLogMgr *GeneralLogManager
+
+// PutGeneralLogOrDrop sends the general log entry to the logging goroutine
+// The entry will be dropped once the channel is full
+func PutGeneralLogOrDrop(entry *GeneralLogEntry) {
+	select {
+	case generalLogMgr.logEntryChan <- entry:
+	default:
+		// When logEntryChan is full, the system resource is under so much pressure that the logging system capacity
+		// is reduced. We should NOT output another warning log saying that we are dropping a general log, which will
+		// add more pressure on the system load.
+		// TODO(dragonly): add prometheus counter for dropped log count?
```

Comment:
Agree. I think we should add a counter metric for the droppedLogCount so that user can know their log have been dropped by the metric.